web/packages/extension/src/background.ts

```diff
@@ -8,7 +8,138 @@ async function contentScriptRegistered() {
     return matchingScripts?.length > 0;
 }
 
+async function isHeaderConditionSupported() {
+    let needCleanup = false;
+    const ruleId = 4;
+    try {
+        // Throws synchronously if not supported.
+        await utils.declarativeNetRequest.updateDynamicRules({
+            addRules: [
+                {
+                    id: ruleId,
+                    condition: { responseHeaders: [{ header: "whatever" }] },
+                    action: {
+                        type:
+                            chrome.declarativeNetRequest.RuleActionType
```

Comment:
Should we use `utils.declarativeNetRequest` consistently?

Reply:
And honestly, I don't see a point in doing this instead of just using strings.

Reply:
TS yells about anything else due to type definitions by DefinitelyTyped.

Reply:
See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/223f242577e08bea526d40d53eee9fea9b13a9f1/types/chrome/index.d.ts#L12300 and https://github.com/DefinitelyTyped/DefinitelyTyped/blob/223f242577e08bea526d40d53eee9fea9b13a9f1/types/chrome/index.d.ts#L12196. For chrome this must be a defined enum member.

See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/223f242577e08bea526d40d53eee9fea9b13a9f1/types/webextension-polyfill/namespaces/declarativeNetRequest.d.ts#L486 and https://github.com/DefinitelyTyped/DefinitelyTyped/blob/223f242577e08bea526d40d53eee9fea9b13a9f1/types/webextension-polyfill/namespaces/declarativeNetRequest.d.ts#L413. For browser this must be a string.

Reply:
I think we should probably just use one namespace and stick with at some point in the future.

Reply:
Yeah, but that should ideally be `browser` and Chrome still doesn't support that. For our use-case `chrome` does work fine but the WECG says that's not supposed to be the standard.

Reply:
Once https://crbug.com/798169 is addressed we can just use browser, but we'd have to wait until whatever version of Chromium supports that is old enough to be the oldest version we want to support.

Reply:
Added TODO comment for switching to a single namespace to utils.ts. As stated, I unfortunately can't change this with the current DefinitelyTyped type definitions, so just marking this as resolved.